kroxylicious/src/main/java/io/kroxylicious/proxy/bootstrap/FilterChainFactory.java

```diff
@@ -25,8 +28,23 @@ public class FilterChainFactory {
     private final Configuration config;
 
     public FilterChainFactory(Configuration config) {
-        Objects.requireNonNull(config);
-        this.config = config;
+        this.config = Objects.requireNonNull(config);
+    }
+
+    /**
+     * Validate that a collection of FilterDefinitions contains all the required configuration to configure the filter.
+     * @param filterDefinitions the candidates to validate.
+     * @return the Set of filter names which fail validation.
+     */
+    public static Set<String> validateFilterConfiguration(Collection<FilterDefinition> filterDefinitions) {
+        if (Objects.isNull(filterDefinitions) || filterDefinitions.isEmpty()) {
```

Comment:
minor preference for `filterDefinitions == null` here, Objects::isNull looks like it's mostly for use as a Predicate

Reply:
Personally I'd like to be able to assume its never null, but given an opinion has been expressed I'll update it.

Reply:
resolved with ba4a8815eb407a607a4b6a5f5b0ce9fc091e28be